lib/engine/game/g_1812/game.rb

```diff
@@ -42,69 +42,341 @@ class Game < Game::Base
+        def post_train_buy; end
+
+        def new_or!
+          if @round.round_num < @operating_rounds
+            new_operating_round(@round.round_num + 1)
           else
-            self.class::MINORS
+            @turn += 1
+            or_set_finished
+            new_stock_round
           end
         end
 
-        # def bank_sort(corporations)
-        #   minors, corps = entities.partition(&:minor?)
-        #   minors.sort_by { |m| m.name.to_i } + super(corps)
-        # end
+        def next_round!
+          clear_interest_paid
+          @round =
+            case @round
+            when Engine::Round::Stock
+              @operating_rounds = @phase.operating_rounds
+              reorder_players
+              new_operating_round
+            when Engine::Round::Operating
+              or_round_finished
+              if phase.name.to_i == 2
+                new_or!
+              else
+                @log << "-- #{round_description('Merger', @round.round_num)} --"
+                G1867::Round::Merger.new(self, [
+                G1867::Step::ReduceTokens,
+                Engine::Step::DiscardTrain,
+                G1867::Step::Merge,
+              ], round_num: @round.round_num)
+              end
+            when G1867::Round::Merger
+              new_or!
+            when init_round.class
+              reorder_players
+              new_stock_round
+            end
+        end
+
+        def or_set_finished
+          depot.export_all!('2') if @phase.name.to_i == 2 &&
+                                    (@depot.upcoming.first != train_by_id('2-0') &&
+                                     @depot.upcoming.first != train_by_id('3-0'))
+
+          return unless @convert_3s == true
+
+          # Count the remaining '3' trains
+          remaining_trains_count = depot.upcoming.count { |t| t.name == '3' }
+          return unless remaining_trains_count
+
+          # Forget all remaining '3' trains
+          trains = depot.upcoming.select { |t| t.name == '3' }
+          trains.each { |t| depot.forget_train(t) }
+
+          # Increase the num value of the '3+1' trains by the stored count
+          replace_three_with_three_plus_one(remaining_trains_count)
+          @convert_3s = false
+        end
+
+        def replace_three_with_three_plus_one(remaining_trains_count)
+          @log << '--All remaining 3/2G trains in the supply replaced with with 3+1/2G+1 trains--'
+
+          remaining_trains_count.times do
+            train = @three_plus_one.shift
+            train.reserved = false
+            @depot.unshift_train(train)
+          end
+        end
+
+        def init_train_handler
+          depot = super
+
+          # store the 3+1 trains in reserve for now
+
+          @three_plus_one = []
+          5.times do
+            train = depot.upcoming.find { |t| t.name == '3+1' }
+            @three_plus_one << train
+            depot.remove_train(train)
+            train.reserved = true
+          end
```

Comment:
You are stashing all of the 3+1 trains, aren't you? So you don't need to hardcode the '5' quantity. And I think this could be a little simpler too.
```suggestion
          @three_plus_one = depot.upcoming.select { |t| t.name == '3+1' }
          @three_plus_one.each do |train|
            depot.remove_train(train)
            train.reserved = true
          end
```

Reply:
done